### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by Btibert3** [hiratake55/RForcecom#16]
Any chance that you might add some functionality to work with the metadata API?

**Comment by Btibert3** [comment]
Is the creating/deleting objects and fields even possible?  It would be great to be able to programmatically, with R, build out orgs quickly to demonstrate proofs of concept.

**Comment by ReportMort** [comment]
@Btibert3 Yes, you should be able to create/delete objects and fields since the Metadata API supports  [CRUD operations.] (https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_crud_based_calls_intro.htm) I'll be taking a look at the Metadata API soon.

**Comment by Btibert3** [comment]
Awesome.  I am currently trying to work through the code, but admittedly I am still trying to figure out how it might work through R.  That said, if you do get something hacked together, I am pretty sure that I could run with an example and would absolutely love to help out on that side of the equation.

**Comment by ReportMort** [comment]
Below is a link to a gist that demos reading the metadata from the Account object, then switching the name on it to "Custom_Account__c", and re-submiting the metadata back into Salesforce to create a custom copy of the Account object. The code is still really rough and we'll need to write some convenience wrappers to abstract a lot of the action, but figured it'd help get the ball rolling.

https://gist.github.com/ReportMort/f84f6d853ea3c886ebe4

**Comment by Btibert3** [comment]
That is fantastic.  Let me try to wrap my head around it and see what I can do.  Thanks for kicking this off!
